### ISSUE TITLE
feat: Support loading from 

### ISSUE THREAD

**Issue by dwieeb** [ionic-team/capacitor#3912]

**Comment by rraihansaputra** [comment]
I'd like to share another usecase, currently I'm building a Capacitor app with Svelte and Routify, and for SSR purposes the default entrypoint is `__app.html`. Currently my workaround for building the Capacitor app both locally and in CI is to have a separate build step for the web app which copies `dist/__app.html` to `dist/index.html` prior to executing `npx cap copy`. If this can be configured, it would resolve the separate jobs needed to make this work.

**Comment by sabahallah** [comment]
Dear @imhoffd 

Any luck when this could be resolved.

We now architecting an application that will use Angular built-in i18n for multi-lingual app. and with using Angular built-in i18n there would be a separate build for each language. We will use Capacitor as a hybrid mobile solution, but Capacitor copy the application from www directory, in my case there will be folder for each language. So how to handle this in Capacitor, and is it possible to change the URL in runtime as the user change the language.

**Comment by meriturva** [comment]
Any news here? we would like to use angular i18n and we have to decide how to proceed. Thanks!

**Comment by UltraCakeBakery** [comment]
There is currently no way to do native translations that allow a user to only install for example the english version of your app in your app bundle. 

You do however have access in your web app to https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages, which allows you to determine the devices preferred language.

**Comment by meriturva** [comment]
Thanks @UltraCakeBakery for your feedback.

My idea is to pack capacitor with all language folders produced by ng build and add an index.html file on the root folder.
The root index file has just to check the device's preferred language and redirects to the correct folder.

What do you think?

**Comment by UltraCakeBakery** [comment]
> Thanks @UltraCakeBakery for your feedback.
> 
> My idea is to pack capacitor with all language folders produced by ng build and add an index.html file on the root folder. The root index file has just to check the device's preferred language and redirects to the correct folder.
> 
> What do you think?

That'll work fine I think. It is all local so redirects won't hurt performance too much.

**Comment by rdlabo** [comment]
> Thanks @UltraCakeBakery for your feedback.
> 
> My idea is to pack capacitor with all language folders produced by ng build and add an index.html file on the root folder. The root index file has just to check the device's preferred language and redirects to the correct folder.
> 
> What do you think?

This work well. I prepared this html:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <script type="text/javascript">
      // If true, you should set localStorage[storageKey] in your app.
      const saveLocale = true;
      const storageKey = 'locale';
      const defaultLocale = 'ja';
      const localeMap = {
        ja: ['ja', 'ja-JP'],
        'en-us': ['en', 'en-US'],
      };

      // Method
      const getBrowserLanguage = (navigator) => {
        if (!navigator) return null;
        if (navigator.languages && navigator.languages.length > 0) return navigator.languages[0];
        if (navigator.userLanguage) return navigator.userLanguage;
        return navigator.language;
      };
      const getToRoute = (locale) => {
        const matchRoute = Object.keys(localeMap).find((key) => {
          if (localeMap[key].includes(locale)) return true;
        });
        return `${matchRoute}`;
      };

      // Run redirect
      (() => {
        let locale = saveLocale === true ? localStorage.getItem(storageKey) : null;
        if (!locale) locale = getBrowserLanguage(window.navigator);
        if (saveLocale === true) localStorage.setItem(storageKey, locale);
        if (!locale) locale = defaultLocale;
        const route = getToRoute(locale.toLowerCase());
        window.location.replace([new URL(window.location.href).origin, route, 'index.html'].join('/'));
      })();
    </script>
  </head>
</html>
```

**Comment by Erudition** [comment]
Yes, it seems that setting `server: url` to ANYTHING (including the default, `https://localhost/`) prevents all localhost paths from working (Err Connection Refused). Would love to start somewhere else without losing the local server...

**Comment by Erudition** [comment]
Here's my current workaround, what do y'all think.
Added a (p)npm script:
```json
"capacitor:copy:after": "if [ $CAPACITOR_PLATFORM_NAME == 'web' ]; then echo leaving index in place for web. ; else echo replacing index with go-online... && mv android/app/src/main/assets/public/index.html android/app/src/main/assets/public/original.html && mv android/app/src/main/assets/public/go-online.html android/app/src/main/assets/public/index.html; fi"
```
Which moves the desired page to index.html for mobile platforms only.

**Comment by selected-pixel-jameson** [comment]
Would be great to see some movement on this. Ionic w/Angular standalone is pretty much dead in the water for localization options seeing as ngx-translate doesn't support standalone. With the advancement of AI it's only going to make more sense for solutions to incorporate localization since it is now becoming so affordable and easy to translate text. I only see the need for Ionic to support Angular's native localization to become more and more prevalent. 

Even a solution that would let us give the end user an interface to toggle between, which would restart the entire application would be acceptable in my opinion, as long as it works with Angular's native translation packages, which are actually awesome.

**Comment by Saqib92** [comment]
it means Capacitor still not have ability to detect other file name such as `index.csr.html` for angular ssr.

**Comment by WhatsThatItsPat** [comment]
I don't think this should be closed. The issue is about doing this dynamically during runtime, specifically for localization. The new `appStartPath` is more like "configure time" or "sync time" feature. We want to be able to reboot the app with a different `appStartPath` to take advantage of Angular's build-time localization strategy.

**Comment by Saqib92** [comment]
Capacitor is falling behind in this type of support. We need something more flexible. we need support for SSR with capacitor
